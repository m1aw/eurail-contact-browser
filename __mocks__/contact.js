export const contactMock = {
    "gender": "female",
    "name": {
        "title": "Ms",
        "first": "Loane",
        "last": "Mercier"
    },
    "location": {
        "street": {
            "number": 1826,
            "name": "Boulevard de Balmont"
        },
        "city": "NÃ®mes",
        "state": "Cher",
        "country": "France",
        "postcode": 92735,
        "coordinates": {
            "latitude": "2.0946",
            "longitude": "35.9945"
        },
        "timezone": {
            "offset": "+9:30",
            "description": "Adelaide, Darwin"
        }
    },
    "email": "Loane.Mercier@example.com",
    "login": {
        "uuid": "9ace22a2-eb1c-4a3a-99f1-d4949847ef13",
        "username": "beautifulsnake364",
        "password": "virginie",
        "salt": "SdXNnOtL",
        "md5": "aec794b5992ec92967eca0fbaf945be5",
        "sha1": "feaef2a39c6e9871a3f9a33591411cddd017a108",
        "sha256": "76edff81182948ea707b98066303bf841579796049310043f3119903c4b5da7a"
    },
    "dob": {
        "date": "1975-05-17T00:40:51.032Z",
        "age": 44
    },
    "registered": {
        "date": "2006-08-20T21:17:39.206Z",
        "age": 13
    },
    "phone": "03-24-99-12-06",
    "cell": "06-99-28-89-03",
    "id": {
        "name": "INSEE",
        "value": "2NNaN99001462 16"
    },
    "picture": {
        "large": "https://randomuser.me/api/portraits/women/58.jpg",
        "medium": "https://randomuser.me/api/portraits/med/women/58.jpg",
        "thumbnail": "https://randomuser.me/api/portraits/thumb/women/58.jpg"
    },
    "nat": "FR"
}
export const contactMock2 =  {
    "gender": "male",
    "name": {
        "title": "Mr",
        "first": "Arnold",
        "last": "Edwards"
    },
    "location": {
        "street": {
            "number": 2415,
            "name": "Forest Ln"
        },
        "city": "Sioux Falls",
        "state": "Arizona",
        "country": "United States",
        "postcode": 46902,
        "coordinates": {
            "latitude": "88.1574",
            "longitude": "-111.5056"
        },
        "timezone": {
            "offset": "+9:00",
            "description": "Tokyo, Seoul, Osaka, Sapporo, Yakutsk"
        }
    },
    "email": "Arnold.Edwards@example.com",
    "login": {
        "uuid": "13e34d06-3c8c-4ada-8423-d74e4b81a569",
        "username": "blackduck879",
        "password": "dawg",
        "salt": "oXFVQ2NF",
        "md5": "0958ae902fdbd362f0e4776007e7eeda",
        "sha1": "6243a80c7f11e207c51d9715a178310e95f1a68c",
        "sha256": "82453bdb4b0f2b3390a0973a986b0281f857926312e3e1a2a4bfba36e2eda388"
    },
    "dob": {
        "date": "1970-05-26T14:26:27.822Z",
        "age": 49
    },
    "registered": {
        "date": "2007-04-02T22:25:47.631Z",
        "age": 12
    },
    "phone": "(365)-163-3583",
    "cell": "(476)-642-4385",
    "id": {
        "name": "SSN",
        "value": "907-51-4901"
    },
    "picture": {
        "large": "https://randomuser.me/api/portraits/men/89.jpg",
        "medium": "https://randomuser.me/api/portraits/med/men/89.jpg",
        "thumbnail": "https://randomuser.me/api/portraits/thumb/men/89.jpg"
    },
    "nat": "US"
}