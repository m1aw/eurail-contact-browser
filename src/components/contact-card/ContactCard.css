.ContactCard {
    border-radius: 15px 15px;
    text-align: left;
    border: #CCCCCC solid 1px;
    background: #EEE;
    padding: 20px;
    position: absolute;
    display: flex;
    flex-wrap: wrap;
    min-width: 100%;
    z-index: 10;
}

.ContactCard .info {
    flex: 70%;
}

.Avatar {
    flex: 20%;
}

.ContactCard table {
    font-size: 0.7rem;
}

.ContactCard-label{
    font-weight: bold;
}

.ContactCard-close-button {
    position: absolute;
    top: 10px;
    left: 10px;
    color: #999;
}

.ContactCard-username {
    margin-top: -5px;
    padding: 4px;
    color: #ffffff;
    font-size: 0.7rem;
    position: absolute;
    transform: rotate(90deg);
    top: 0;
    transform-origin: left top;
    left: 98%;
    white-space: nowrap;
    background: rgb(255,34,0);
    background: linear-gradient(90deg, rgba(255,34,0,1) 0%, rgba(255,175,0,1) 32%, rgba(255,89,0,1) 100%);
    box-shadow: 0 0 3px rgba(0, 0, 0, 0.12), 2px 1px 2px rgba(0, 0, 0, 0.24);
}