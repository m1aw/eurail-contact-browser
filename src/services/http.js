import qs from 'qs'

export class HttpService {

    defaultOptions = {
        headers: {
            'Content-Type': 'application/json',
        }
    }

    get = (url, params, config) => {
        let getUrl = url;
        if (params) {
            getUrl = url + '?' + qs.stringify(params);
        }
        return this.request('GET', getUrl, config);
    };

    post = (url, data, config) => this.request('POST',
        url,
        {
            body: JSON.stringify(data),
            ...config,
        });

    put = (url, data, config) => this.request('PUT',
        url,
        {
            body: JSON.stringify(data),
            ...config,
        });

    delete = (url, data, config) => this.request('DELETE',
        url,
        {
            body: JSON.stringify(data),
            ...config,
        });


    request(method, url, opts) {
        const options = {
            method,
            ...this.defaultOptions,
            ...opts,
        }

        return fetch(url, options)
            .then(this.handleErrors)
            .then(res => {
                if (res.status === 204 || res.status === 201) {
                    return res
                }

                return res.json()
            })
            .catch((err) => {
                throw err
            })
    }

    async handleErrors(response) {
        if (response.ok) {
            return response
        }

        throw new Error(response.text)
    }

}