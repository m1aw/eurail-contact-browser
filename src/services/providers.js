import React from "react";

export const ExpandedContactContext = React.createContext();