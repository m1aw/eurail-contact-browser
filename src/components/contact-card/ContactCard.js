import React, {useContext} from 'react'
import './ContactCard.css'
import {ExpandedContactContext} from "../contact-list/ContactList";
import {Avatar} from "../avatar/Avatar"
import fontawesome from '@fortawesome/fontawesome'
import {FontAwesomeIcon} from "@fortawesome/react-fontawesome";
import {faTimes} from "@fortawesome/free-solid-svg-icons";


fontawesome.library.add(faTimes);

export const ContactCard = ({contact}) => {
    const [, setSelectedContact] = useContext(ExpandedContactContext);

    const handleClose = () => setSelectedContact(undefined)

    return (
        <div className={'ContactCard'}>
            <button className={'ContactCard-close-button'} onClick={handleClose}>
                <FontAwesomeIcon icon="times" />
            </button>
            <Avatar contact={contact}/>
            <div className={'info'}>
                <div className={'row'}>
                    <span className={'ContactItem-surname'}>{contact.name.last}</span>, {contact.name.first}
                </div>
                <table>
                    <tbody>
                    <tr className={'row'}>
                        <th className={'ContactCard-label'}>e-mail</th>
                        <td>{contact.email}</td>
                    </tr>
                    <tr className={'row'}>
                        <th className={'ContactCard-label'}>phone</th>
                        <td>{contact.phone}</td>
                    </tr>
                    <tr className={'row'}>
                        <th className={'ContactCard-label'}>street</th>
                        <td>{contact.location.street.name} {contact.location.street.number}</td>
                    </tr>
                    <tr className={'row'}>
                        <th className={'ContactCard-label'}>city</th>
                        <td>{contact.location.city}</td>
                    </tr>
                    <tr className={'row'}>
                        <th className={'ContactCard-label'}>state</th>
                        <td>{contact.location.state}</td>
                    </tr>
                    <tr className={'row'}>
                        <th className={'ContactCard-label'}>postcode</th>
                        <td>{contact.location.postcode}</td>
                    </tr>
                    </tbody>
                </table>
            </div>
            <div className={'ContactCard-username'}>
                USERNAME {contact.login.username}
            </div>
        </div>
    )
}