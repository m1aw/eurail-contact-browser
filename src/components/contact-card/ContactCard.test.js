import React from 'react';
import renderer from 'react-test-renderer';
import {ContactCard} from "./ContactCard";
import {contactMock} from '../../../__mocks__/contact'
import {ExpandedContactContext} from "../../services/providers";

test('ContactCard should match snapshot', () => {
    const mockStateFn = jest.fn();
    const element = renderer.create(
        <ExpandedContactContext.Provider value={[null, mockStateFn]}>
            <ContactCard contact={contactMock} />
        </ExpandedContactContext.Provider>
    );
    let tree = element.toJSON();
    expect(tree).toMatchSnapshot();
    expect(mockStateFn).not.toBeCalled();
    //click on the close button and expect state to change
    element.root.findByType('button').props.onClick();
    expect(mockStateFn).toBeCalled();
    expect(tree).toMatchSnapshot();
});
