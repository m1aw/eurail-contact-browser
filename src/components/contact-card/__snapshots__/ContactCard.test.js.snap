// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactCard should match snapshot 1`] = `
<div
  className="ContactCard"
>
  <button
    className="ContactCard-close-button"
    onClick={[Function]}
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-times fa-w-11 fa-null fa-rotate-null fa-pull-null "
      data-icon="times"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 352 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
  </button>
  <div
    className="Avatar"
  >
    <img
      alt="contact avatar"
      src="https://randomuser.me/api/portraits/med/women/58.jpg"
    />
  </div>
  <div
    className="info"
  >
    <div
      className="row"
    >
      <span
        className="ContactItem-surname"
      >
        Mercier
      </span>
      , 
      Loane
    </div>
    <table>
      <tbody>
        <tr
          className="row"
        >
          <th
            className="ContactCard-label"
          >
            e-mail
          </th>
          <td>
            Loane.Mercier@example.com
          </td>
        </tr>
        <tr
          className="row"
        >
          <th
            className="ContactCard-label"
          >
            phone
          </th>
          <td>
            03-24-99-12-06
          </td>
        </tr>
        <tr
          className="row"
        >
          <th
            className="ContactCard-label"
          >
            street
          </th>
          <td>
            Boulevard de Balmont
             
            1826
          </td>
        </tr>
        <tr
          className="row"
        >
          <th
            className="ContactCard-label"
          >
            city
          </th>
          <td>
            Nîmes
          </td>
        </tr>
        <tr
          className="row"
        >
          <th
            className="ContactCard-label"
          >
            state
          </th>
          <td>
            Cher
          </td>
        </tr>
        <tr
          className="row"
        >
          <th
            className="ContactCard-label"
          >
            postcode
          </th>
          <td>
            92735
          </td>
        </tr>
      </tbody>
    </table>
  </div>
  <div
    className="ContactCard-username"
  >
    USERNAME 
    beautifulsnake364
  </div>
</div>
`;

exports[`ContactCard should match snapshot 2`] = `
<div
  className="ContactCard"
>
  <button
    className="ContactCard-close-button"
    onClick={[Function]}
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-times fa-w-11 fa-null fa-rotate-null fa-pull-null "
      data-icon="times"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 352 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
  </button>
  <div
    className="Avatar"
  >
    <img
      alt="contact avatar"
      src="https://randomuser.me/api/portraits/med/women/58.jpg"
    />
  </div>
  <div
    className="info"
  >
    <div
      className="row"
    >
      <span
        className="ContactItem-surname"
      >
        Mercier
      </span>
      , 
      Loane
    </div>
    <table>
      <tbody>
        <tr
          className="row"
        >
          <th
            className="ContactCard-label"
          >
            e-mail
          </th>
          <td>
            Loane.Mercier@example.com
          </td>
        </tr>
        <tr
          className="row"
        >
          <th
            className="ContactCard-label"
          >
            phone
          </th>
          <td>
            03-24-99-12-06
          </td>
        </tr>
        <tr
          className="row"
        >
          <th
            className="ContactCard-label"
          >
            street
          </th>
          <td>
            Boulevard de Balmont
             
            1826
          </td>
        </tr>
        <tr
          className="row"
        >
          <th
            className="ContactCard-label"
          >
            city
          </th>
          <td>
            Nîmes
          </td>
        </tr>
        <tr
          className="row"
        >
          <th
            className="ContactCard-label"
          >
            state
          </th>
          <td>
            Cher
          </td>
        </tr>
        <tr
          className="row"
        >
          <th
            className="ContactCard-label"
          >
            postcode
          </th>
          <td>
            92735
          </td>
        </tr>
      </tbody>
    </table>
  </div>
  <div
    className="ContactCard-username"
  >
    USERNAME 
    beautifulsnake364
  </div>
</div>
`;
